packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
+    public interval: number = 50;
+
+    constructor(public readonly maxWaitTime = 5000) {
+        /*
+        In the extending class, call the following in the constructor
+         this.waitForAnalytics();
+         this.waitForEvolv();
+        */
+    }
+
+    abstract getAnalytics(): any;
```

Comment:
`getAnalytics()` does appear to be used only as a function later on

```suggestion
    abstract getAnalytics(): (...args: unknown[]) => void;
```